code/game/atoms_movable.dm

```diff
@@ -272,3 +275,20 @@
 	if (src.master)
 		return src.master.attack_hand(a, b, c)
 	return
+
+/atom/movable/proc/handle_buckled_mob_movement(loc,direct)
+	buckled_mob.buckled = null
+	if(!buckled_mob.Move(loc, direct))
+		buckled_mob.buckled = src
+		loc = buckled_mob.loc
+		last_move = buckled_mob.last_move
+		inertia_dir = last_move
+		buckled_mob.inertia_dir = last_move
+		. = 0
+	buckled_mob.buckled = src

```

Comment:
why did you readd this trick of unbuckling the mob, see if they can move and rebuckle them? It causes issues with the janicart so I removed it.